File: scenarios/java/java-concurrent-queue/template/src/main/java/TaskQueue.java
package main.java;

import java.time.Duration;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Callable;

public class TaskQueue<T> {

    public static class RateLimit {
        public int tokens;
        public Duration interval;
        public RateLimit(int t, Duration i) { tokens = t; interval = i; }
    }

    public static class Options {
        public int concurrency;
        public RateLimit rateLimit;
        public int maxRetries;
        
        public Options(int c, RateLimit r, int m) {
            concurrency = c; rateLimit = r; maxRetries = m;
        }
    }

    public TaskQueue(Options options) {
    }

    public CompletableFuture<T> add(Callable<T> task) {
        throw new RuntimeException("Not implemented");
    }
}

File: scenarios/php/php-concurrent-queue/template/src/TaskQueue.php
<?php

class TaskQueue {
    public function __construct(int $concurrency, array $rateLimit = [], int $maxRetries = 0) {
        // ...
    }

    public function add(callable $task) {
        // ...
    }

    public function run(): array {
        return [];
    }
}
